web/packages/teleterm/src/services/config/appConfigSchema.ts

```diff
@@ -338,3 +339,14 @@ function getDefaultTerminalFont(platform: Platform) {
 function getShortcutDesc(actionDesc: string): string {
   return `Shortcut to ${actionDesc}. A valid shortcut contains at least one modifier and a single key code, for example "Shift+Tab". Function keys do not require a modifier.`;
 }
+
+z.config({
+  customError: iss => {
+    switch (iss.code) {
+      case 'invalid_type':
+        return `Expected ${iss.expected}, received ${typeof iss.input}`;
```

Comment:
In practice I don't think we'll have that many codes to address to get something like the old behavior.

`typeof iss.input` is kind of a hack, but it should work since we operate on JSON only. I need to check how the situation looks like for enums.

Reply:
I’m not sure if we'll need custom errors. I’ve learned two things:
* zod v4 prints more detailed messages only if locale is set. 
  After I set `z.config(z.locales.en())`, "Invalid input" becomes "Invalid input: expected boolean, received string". I'm not a fan of that "invalid input" part, but I think we can live with it.
* zod v4 by default doesn't report the input value. However, the `parse()` function accepts `{ reportInput }` parameter, so we can retrieve it along with the error message. I believe it's more useful to the user than only "received bool, expected string" message.

Reply:
I've seen `"Invalid input: expected boolean, received string"` in tests even without setting a locale. I also didn't like the "Invalid input" part, hence this change under which we're commenting on that gets rid of it.

The input value isn't reported by default, but the global config map does have access to it (as also shown by this change), even without passing `reportInput: true`.